planner/optimize.go

```diff
@@ -73,7 +73,14 @@ func GetExecuteForUpdateReadIS(node ast.Node, sctx sessionctx.Context) infoschem
 			execID = vars.PreparedStmtNameToID[execStmt.Name]
 		}
 		if preparedPointer, ok := vars.PreparedStmts[execID]; ok {
-			if preparedObj, ok := preparedPointer.(*core.CachedPrepareStmt); ok && preparedObj.ForUpdateRead {
+			// RC isolation level or statement with for-update read need to check if the schema is changed
+			// before use the cached plan.
+			checkSchema := vars.IsIsolation(ast.ReadCommitted)
```

Comment:
Should prepare/execute in binary protocol be considerate too?

Reply:
Yes, here is the code path:
![image](https://user-images.githubusercontent.com/7499936/168008759-826e9df5-b97a-48a1-80d8-d2b9173b6976.png)